### PR TITLE
Fix CMake target include directories

### PR DESCRIPTION
**Pull request by r4nx**
Include directory is not set if the library is used with `add_subdirectory` or `FetchContent`, this PR fixes it

### CODE REVIEW

File: c/CMakeLists.txt [hunk 1]
```diff
@@ -47,7 +47,10 @@ if (BUILD_SHARED_LIBS)
   )
   list(APPEND BLAKE3_PKGCONFIG_CFLAGS -DBLAKE3_DLL)
 endif()
-target_include_directories(blake3 PUBLIC $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_include_directories(blake3 PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
```
Comment (BurningEnlightenment) [inline c/CMakeLists.txt:51]:
```suggestion
  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
```
Less stuff to change if we ever split the `CMakeLists.txt`.

**Review by BurningEnlightenment — CHANGES_REQUESTED** [review]
Apart from the suggestion I'm okay with this.

**Comment by r4nx** [comment]
> Apart from the suggestion I'm okay with this.

Thanks, sure, I've commited the changes

**Review by BurningEnlightenment — APPROVED** [review]
@oconnor663 LGTM

**Pull request merged**